contracts/Marketplace.sol

```diff
@@ -67,12 +68,14 @@ contract Marketplace is Collateral, Proofs {
 
     slot.host = msg.sender;
     slot.requestId = requestId;
-    RequestContext storage context = requestContexts[requestId];
+    RequestContext storage context = _context(requestId);
     context.slotsFilled += 1;
+    context.endsAt = block.timestamp + request.ask.duration;
```

Comment:
A request should end at `block.timestamp + request.ask.duration` in `requestStorage()`, not in`fillSlot()`. The end time of the contract is determined at request submission, it is not dependent on when slots are filled or when the contract is started.

Reply:
Each host will need to provide proofs for `request.ask.duration`, correct? This would mean that when the last slot is filled, the request would be active until `block.timestamp + request.ask.duration`.

Otherwise, if the request ended at `time_requested + request.ask.duration`, that means that the host filling the last slot would only need to provide proofs for less than `request.ask.duration`.

Reply:
No, hosts only need to provide proofs from the moment they fill a slot until the contract ends. That is indeed shorter for hosts that fill a slot later in time. But they also earn less money because they get paid for the amount of time that they fill a slot.

Reply:
All updated.